showcase/app/templates/components/table.hbs

```diff
@@ -142,6 +141,72 @@
     </:body>
   </Hds::Table>
 
+  {{! ================================================== }}
+  {{! ===== DEMO #5 - SORT BY SELECTED ===== }}
+  {{! ================================================== }}
+
+  <Shw::Text::H4 @tag="h3">Sortable table with sorting by selected item</Shw::Text::H4>
+
+  <Hds::Table
+    @isSelectable={{true}}
+    @sortBySelectedItemKey="isSelected"
+    @onSelectionChange={{this.onSelectionChange__demo5}}
+    @model={{this.selectableData__demo5}}
+    @columns={{array
+      (hash key="lorem" label="Row #" isSortable=true)
+      (hash key="ipsum" label="Ipsum")
+      (hash key="dolor" label="Dolor")
+    }}
+  >
+    <:body as |B|>
+      <B.Tr
+        @selectionKey={{B.data.id}}
+        @isSelected={{B.data.isSelected}}
+        @selectionAriaLabelSuffix="row #{{B.data.lorem}}"
+      >
+        <B.Td>{{B.data.lorem}}</B.Td>
+        <B.Td>{{B.data.ipsum}}</B.Td>
+        <B.Td>{{B.data.dolor}}</B.Td>
+      </B.Tr>
+    </:body>
+  </Hds::Table>
+
+  <Shw::Text::H4 @tag="h3">Sortable table with no model, sorting by selected item</Shw::Text::H4>
+  <Shw::Text::Body>⚠️
+    <em>Notice: in this example the "sort" button doesn't work, is for presentation purpose only.</em></Shw::Text::Body>
+
+  <Hds::Table
+    @caption="a sortable table with no model defined that can sort by selection state"
+    @isSelectable={{true}}
+    @onSelectionChange={{this.onSelectionChangeSortBySelectedWithoutModel__demo6}}
```

Comment:
[todo] since this example is "for presentation only" we don't need this custom "sorting" function (you could use `@onSelectionChange={{this.onSelectionChangeLogArguments}}` like on other places; if instead the intent is to validate that it's possible to sort by selected row even when there's no `@model/@columns` then the example should be moved under the "Custom sorting" section, but in that case the logic/code should be functional

Reply:
I'm not certain we need to validate that it's possible as almost all of the logic for implementing it will be outside of the component. I think we've already validated elsewhere that the sortProperty will be passed to the custom onSort function, but I can add a functional example with custom sort by selected behavior if you think it will add value.

Reply:
no, I don't think we need to test that use case (the logic is client-side, and as you say above we know the sortProperty is passed to the custom onSort function).